src/main/java/de/tum/in/www1/artemis/web/rest/admin/AdminBuildJobQueueResource.java

```diff
@@ -66,4 +64,50 @@ public ResponseEntity<List<LocalCIBuildAgentInformation>> getBuildAgentInformati
         List<LocalCIBuildAgentInformation> buildAgentInfo = localCIBuildJobQueueService.getBuildAgentInformation();
         return ResponseEntity.ok(buildAgentInfo);
     }
+
+    /**
+     * Cancels the build job for the given participation.
+     *
+     * @param buildJobId the id of the build job to cancel
+     * @return the ResponseEntity with the result of the cancellation
+     */
+    @DeleteMapping("/cancel/{buildJobId}")
```

Comment:
I don't really like the naming of the endpoint here. api/admin/cancel/{id} does not seem to be descriptive enough
```suggestion
    @DeleteMapping("/cancel-job/{buildJobId}")
```